plotly/tools.py

```diff
@@ -1559,38 +1551,43 @@ def _trisurf(x, y, z, simplices, colormap=None, dist_func=None,
 
         min_mean_dists = np.min(mean_dists)
         max_mean_dists = np.max(mean_dists)
-        facecolor = ([FigureFactory._map_z2color(zz, colormap, min_mean_dists,
-                      max_mean_dists) for zz in mean_dists])
-        ii, jj, kk = FigureFactory._tri_indices(simplices)
+        facecolor = FigureFactory._map_z2color(mean_dists, colormap,
+                                               min_mean_dists, max_mean_dists)
+        ii, jj, kk = zip(*simplices)
 
         triangles = graph_objs.Mesh3d(x=x, y=y, z=z, facecolor=facecolor,
                                       i=ii, j=jj, k=kk, name='')
 
-        if plot_edges is None:  # the triangle sides are not plotted
+        if plot_edges is not True:  # the triangle sides are not plotted
             return graph_objs.Data([triangles])
 
         # define the lists x_edge, y_edge and z_edge, of x, y, resp z
         # coordinates of edge end points for each triangle
         # None separates data corresponding to two consecutive triangles
-        lists_coord = ([[[T[k % 3][c] for k in range(4)]+[None]
-                        for T in tri_vertices] for c in range(3)])
-        if x_edge is None:
-            x_edge = []
-        for array in lists_coord[0]:
-            for item in array:
-                x_edge.append(item)
-
-        if y_edge is None:
-            y_edge = []
-        for array in lists_coord[1]:
-            for item in array:
-                y_edge.append(item)
-
-        if z_edge is None:
-            z_edge = []
-        for array in lists_coord[2]:
-            for item in array:
-                z_edge.append(item)
+        is_none = [ii is None for ii in [x_edge, y_edge, z_edge]]
```

Comment:
If we wanted to be extra careful we could add more checks here (e.g., shape is the same for x_edge, y_edge, z_edge)